Automattic/wp-calypso
# Editor: Make the AddTerm Component more reusable (revert the revert)

I had to revert this PR here because of a prod breakage https://github.com/Automattic/wp-calypso/pull/8588

Here is the revert PR: https://github.com/Automattic/wp-calypso/pull/8643

The editor was breaking if we did not have a local cache (we did not load the taxonomies yet), It should be fixed now (2nd commit here). So make sure to clear all the local cache to test this.

**Testing instructions**

* Clear browser's localstorage
* Create a post
* Make sure, you don't have any JS errors when opening the editor
* Add a new category to this post (using the add term modal)
* Make sure the category is attached to the post

cc @timmyc
